src/device-manager/java/src/nl/Weave/DataManagement/ResourceIdentifier.java

```diff
@@ -0,0 +1,105 @@
+/*
+ *
+ *    Copyright (c) 2019 Google LLC
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Represents Resource Identifier.
+ */
+
+package nl.Weave.DataManagement;
+import android.os.Build;
+import android.util.Log;
+import java.math.BigInteger;
+import java.util.Objects;
+
+public class ResourceIdentifier
+{
+    public ResourceIdentifier()
+    {
+        this.resourceType = RESOURCE_TYPE_RESERVED;
+        this.resourceId = SELF_NODE_ID;
+    }
+
+    public ResourceIdentifier(int resourceType, long resourceId) {
+        this.resourceType = resourceType;
+        this.resourceId = resourceId;
+    }
+
+    private final int resourceType;
+    private final long resourceId;
+
+    public static ResourceIdentifier make(int resourceType, BigInteger resourceId)
+    {
+        ResourceIdentifier resourceIdentifier = new ResourceIdentifier(resourceType, resourceId.longValue());
```

Comment:
Could inline the return, here and the next method. i.e.

```
return new ResourceIdentifier(resourceType, resourceId.longValue());
```

Reply:
done, thanks!